examples/form_data/form_data.rs

```diff
@@ -14,15 +13,16 @@ fn main() {
     });
 
     server.post("/confirmation", middleware!{ |req, res|
-        let mut form_data = String::new();
-        req.origin.read_to_string(&mut form_data).unwrap();
+        let form_data = try_with!(res, req.form_body());
 
-        println!("{}", form_data);
+        println!("{:?}", form_data);
 
         let mut data = HashMap::new();
         data.insert("title", "Confirmation");
-        data.insert("formData", &form_data);
-
+        data.insert("firstname", form_data.get("firstname").unwrap_or("What's your first name?"));
+        data.insert("lastname", form_data.get("lastname").unwrap_or("What's your last name?"));
+        data.insert("phone", form_data.get("phone").unwrap_or("Give me your phone or I will give you mine!"));
+        data.insert("email", form_data.get("email").unwrap_or("I want your email!"));
```

Comment:
> Maybe we can add a method require_keys()

I don't think that is a good idea. It would be really nice to get this working with serialization as you pointed above:
```rust
let foo: SomeType = try_with!(req, form_data.parse())
```